util/cache.go

```diff
@@ -0,0 +1,111 @@
+// Handle the password caching
+package util
+
+import (
+	"errors"
+	"time"
+
+	"github.com/patrickmn/go-cache"
+)
+
+type CachedUser struct {
+	Password   string
+	UsageCount int
+	LoggedIn   bool
+}
+
+type VaultCache struct {
+	store *cache.Cache
+}
+
+var UserVaultCache *VaultCache
+
+func NewVaultCache() *VaultCache {
+	var vc *VaultCache
+	vc = new(VaultCache)
+	vc.store = cache.New(cache.NoExpiration, time.Minute*5)
+	return vc
+}
+
+func (vc *VaultCache) Get(User string) (*CachedUser, error) {
+	var cu *CachedUser
+	item, exists := vc.store.Get(User)
+	if exists == false {
+		return nil, errors.New("User not cached")
+	}
+	cu = item.(*CachedUser)
+	return cu, nil
+}
+
+func (vc *VaultCache) AddEntry(User string, Password string) error {
+	cu, err := vc.Get(User)
+	if err == nil {
+		if cu.Password != Password || cu.LoggedIn == false {
```

Comment:
`!cu.LoggedIn`
Please check for other places in this file where you do the same